lib/firefox/webdriver/firefoxDelegate.js

```diff
@@ -26,7 +26,20 @@ class FirefoxDelegate {
     this.hars = [];
   }
 
-  async onStartIteration() {}
+  async onStartIteration(runner) {
+    if (this.firefoxConfig.geckoProfiler) {
```

Comment:
This doesn't support Android (yet!), 'cuz you aren't careful with the output path.  Can you bail if ... oh wait, you can't run Firefox against Android just yet on this branch.

As you were!